File: tournaments/isGeometricProgression/isGeometricProgression.js
function isGeometricProgression(sequence) {
    const n = sequence.length;

    for (let i = 1; n - 1 > i; ++i) {
        if (sequence[i]**2 !== sequence[i-1]*sequence[i+1]) return false;
    }

    return true;
}

File: tournaments/isGeometricProgression/isGeometricProgression.cpp
bool isGeometricProgression(std::vector<int> sequence) {

  for (int i = 1; i < sequence.size(); i++) {
    if (sequence[i] * sequence[0] != sequence[i - 1] * sequence[1]) {
      return false;
    }
  }
  return true;
}
